Replace Flux Oversight Committee

dholbach opened the pull request:
Following up on various discussion threads lately @scottrigby and myself (with the help of other Flux maintainers and TAG Contributor Strategy folks) looked into 

1. replacing the Flux Oversight Committee with governance that is transparently agreed on and closer to how decisions are actually made (The idea behind this was to have a very simple governance change.)
2. bringing our Governance closer to what the Governance WG has put together as recommendation for projects of our structure (e.g. having a maintainer's council): https://github.com/cncf/project-template/blob/main/GOVERNANCE-maintainer.md

While the PR might look a bit long, a lot of pieces of text were copied over from the template as mentioned above and other pieces, e.g. the process definitions into a separate doc called `PROCESS.md`, so we can direct users to one place for "how to interact with the Flux project".

Here is a list of key changes (feel free to check individual commits on this PR to get a better idea of what changed):

- Replace the Oversight Committee (which was not really used in our former governance structure) with the "flux 2 maintainers" as an escalation point. It is currently the group in our project where things are most often decided and the one where members have to demonstrate the deepest involvement beforehand. (The Oversight Committee was decided back in the day, it was never voted in.)
- Code of Conduct: copied from template. It is clearer now and has a section for what to do if the violator is a maintainer.
- Add a section about team meetings (copied as well), so we can define it as an expectation for project members and maintainers.
-  Add a section explaining that almost all decisions are found through (lazy) consensus and that if we can't make decisions, we strive to work with our friends in adjacent communities and CNCF counterpart groups.
- Move process related bits into `PROCESS.md`.
- Expectations for members and maintainers: copy over some lingo from template to broaden our idea of contributions (docs, etc as well) and to give a more concrete idea what "sustained and significant contributions" mean. Set context for 'expected contributions' example - we want to allow for some flexibility. Underline 'ask for help/feedback'.
- Allow for core maintainers to be removed when inactive as well.
- Require 2fa on GitHub from maintainers.
- Further specify unanimity option to prevent a potential future impasse. Decisions can have a deadline date defined by which a quorum of 2/3 is sufficient, but consensus necessary (no veto votes). Define what we mean by Simple Majority and Supermajority
- Define Org Admins team who have full access to the `fluxcd` organisation in GitHub - current members: Flux Oversight Committee + CNCF / LF members.

Follow-up items

- [ ] Rename `@fluxcd/flux2-maintainers` team to `@fluxcd/core-maintainers`
- [ ] Add `@fluxcd/org-admins` team
- [ ] Replace https://github.com/fluxcd/flux2/blob/main/MAINTAINERS content with link to https://github.com/fluxcd/community/blob/main/CORE-MAINTAINERS
- [ ] Update https://github.com/fluxcd/community/tree/main/project to work with new structure
- [ ] Review Org Admin team - see if we have new applicants
- [ ] Set up annual reminder to review Org Admin team
- [ ] Document that the process for reporting CoC violations is to reach out to the private Flux maintainers list


Please leave feedback, we'd love to hear from all of you.

stefanprodan commented on community-roles.md:
```diff
@@ -9,22 +9,15 @@ Roles are progressive, so each include responsibilities, requirements and defini
-* [@fluxcd/oversight-committee](https://github.com/orgs/fluxcd/teams/oversight-committee): The team comprised of all [Oversight Committee] members.
+- [fluxcd org](https://github.com/fluxcd): The organization under which all of Flux's activity on GitHub is captured.
+- [@fluxcd/flux2-maintainers](https://github.com/orgs/fluxcd/teams/flux2-maintainers): The team comprised of all maintainers of Flux v2 components.
+- [@fluxcd/maintainers](https://github.com/orgs/fluxcd/teams/maintainers): The team comprised of all maintainers of both, Flux v1 and Flux v2 components.
```
```suggestion
- [@fluxcd/maintainers](https://github.com/orgs/fluxcd/teams/maintainers): The team comprised of all maintainers of the various projects in the FluxCD organisation: Flux v1, Flux v2, Flagger, GitOps Toolkit, etc.
```

scottrigby commented:
From @jberkus [cncf #tag-contributor-strategy Slack thread](https://cloud-native.slack.com/archives/CT6CWS1JN/p1651018165469609?thread_ts=1650982739.822829&cid=CT6CWS1JN):

❓ @jberkus I checked off the items I believe are addressed – please let us know if you disagree. And added a ❓ emoji in front of follow-up/clarification questions. Thank you!

> - [x] 1. the term Committee is used several times and apparently refers to a specific group, but the term is not defined.

Removed stray references to "committee members" in https://github.com/fluxcd/community/pull/203/commits/20e00c79aec1d3505bfbdebc9e33fec713332c14.

> - [ ] 2. qualifications to be a maintainer are quite lightly defined.  You might consider making them more rigorous; otherwise, it's hard to argue against charges of favoritism when someone gets rejected.  Also, "significant and sustained contributions in a git repository" -- do you use git repos for non-code contributions as well?  Because one of the best ways to expand the diversity of your governing board is to credit non-code contributors and make them eligible to run.

Adopted additional language around membership and maintainership from tag-contributor-strategy, and defined sustained/significant contributions in https://github.com/fluxcd/community/pull/203/commits/ebeffda0a2d538cce178389a7b0747600d2591eb.

Yes, Flux also uses git repos for non-code contributions (fluxcd/community, fluxcd/website which also includes docs). The community role of [Project Member](https://github.com/fluxcd/community/blob/main/community-roles.md#project-member) currently says: 
> "Project Members are [Community Members](https://github.com/fluxcd/community/blob/main/community-roles.md#community-member) who contribute directly to the project and add value to it. This can be through code, documentation, taking part in bug scrubs, etc."

❓ @jberkus does this feel well defined now?

> - [x] 3. There's no clearly defined process for removing a Maintainer.  Consider that you may someday need to do so becuase of a CoC violation.; there needs to be a process that will work in that case, even if the errant maintainer is hostile to being removed.
>         See our template for an example of language you can use

Adopted CoC rules as per suggestion from tag-contributor-strategy: https://github.com/cncf/project-template/blob/main/GOVERNANCE-maintainer.md in https://github.com/fluxcd/community/pull/203/commits/590ff08ebcd553f9e52577dc3e91f180679d10b2.

> - [x] 4. Unanimous votes: You may not have considered the difficulty of holding unanimous votes with a 7-person committee.  Consider the difficultly of holding a vote if someone is out-of-office.  I think this made more sense with a 3-person group.  I suggest a 2/3 majority instead.

Removed the unanimity option to prevent a potential future impasse. Decisions that previously required unanimity now only require supermajority. In https://github.com/fluxcd/community/pull/203/commits/86b50cb6f52592cb3faad06a8a4108e6d3738ffb.

> - [ ] 5. Your other vote mechanisms are defined by a link to wikipedia.  The wikipedia links do not actually define how these votes would be counted in practice.  For example, does one need a simple majority of all maintainers in existence, or just of the maintainers voting?   If the latter, what's the notification process and the time limit?

Replaced those with a definition of Simple Majority and Supermajority in https://github.com/fluxcd/community/pull/203/commits/5c37114ded0ad53edce529eb2fa412955dbe76fa. 

❓ @jberkus does this read more clearly? 

If consensus can not be reached, and a vote is called, the intention is to require either a simple or supermajority of the deciders (depending on the decision, as noted in the "Deciders" section of the doc), not only of those who cast their vote. We did not yet specify the notification process and time limit for decisions generally, because it's not the latter.

❓ @jberkus In practice, Flux maintainers use the public community meetings to inform about decisions in progress and decisions already made. Do you think we should explicitly note this in the governance doc, and/or adopt a different notification process?

> - [x] 6. It would be good to define some kind of community meeting in your governance docs -- the more so if you already have one you're doing.

Added this in https://github.com/fluxcd/community/pull/203/commits/c3cc2e49899c4cf59547b088df72248f053c9c57.

> - [x] 7.  the link for "who to contact" for CoC violations should target your maintainer mailing list.

Added in https://github.com/fluxcd/community/pull/203/commits/590ff08ebcd553f9e52577dc3e91f180679d10b2.

scottrigby commented on community-roles.md:
```diff
@@ -9,22 +9,15 @@ Roles are progressive, so each include responsibilities, requirements and defini
-* [@fluxcd/oversight-committee](https://github.com/orgs/fluxcd/teams/oversight-committee): The team comprised of all [Oversight Committee] members.
+- [fluxcd org](https://github.com/fluxcd): The organization under which all of Flux's activity on GitHub is captured.
+- [@fluxcd/flux2-maintainers](https://github.com/orgs/fluxcd/teams/flux2-maintainers): The team comprised of all maintainers of Flux v2 components.
+- [@fluxcd/maintainers](https://github.com/orgs/fluxcd/teams/maintainers): The team comprised of all maintainers of the various projects in the FluxCD organisation: Flux v1, Flux v2, Flagger, GitOps Toolkit, etc.
```
Reduce possible confusion about whether we also use Git repos for non-code contributions, and whether maintainers of certain projects can focus on non-code contributions (fluxcd/community, fluxcd/website which also includes docs).

Note the community role of [Project Member](https://github.com/fluxcd/community/blob/main/community-roles.md#project-member) does already say this:

> "Project Members are [Community Members](https://github.com/fluxcd/community/blob/main/community-roles.md#community-member) who contribute directly to the project and add value to it. This can be through code, documentation, taking part in bug scrubs, etc."

But it can't hurt to reinforce this message in various places where we mention maintainers by specifically including these repos in the list of examples.

```suggestion
- [@fluxcd/maintainers](https://github.com/orgs/fluxcd/teams/maintainers): The team comprised of all maintainers of the various projects in the FluxCD organisation: Flux v1, Flux v2, Flagger, GitOps Toolkit, website and community repos, etc.
```

jberkus commented:
Re-review!

> > * [x]  1. the term Committee is used several times and apparently refers to a specific group, but the term is not defined.
> 
> Removed stray references to "committee members" in [20e00c7](https://github.com/fluxcd/community/commit/20e00c79aec1d3505bfbdebc9e33fec713332c14).

Those mentions seem to be gone.

> > * [ ]  2. qualifications to be a maintainer are quite lightly defined.  You might consider making them more rigorous; otherwise, it's hard to argue against charges of favoritism when someone gets rejected.  Also, "significant and sustained contributions in a git repository" -- do you use git repos for non-code contributions as well?  Because one of the best ways to expand the diversity of your governing board is to credit non-code contributors and make them eligible to run.
> 
> Adopted additional language around membership and maintainership from tag-contributor-strategy, and defined sustained/significant contributions in [ebeffda](https://github.com/fluxcd/community/commit/ebeffda0a2d538cce178389a7b0747600d2591eb).

So, while I feel that the definition needs more work -- because you're going to need to justify why person#1 is a maintainer and person#2 is not -- I also think you should adopt the governance as-is and then refine the definition of maintainer later.

Here's the ambiguities I currently see in the definition:

1. "5 reviews" Per year?  Per month?  Ever?
2. Usually projects tie a concept of ownership to being a maintainer.  That is, someone is eligible to be a maintainer if they take long-term responsibility for a piece of the project.  That's absent here.
3. Do you want to require flux2 maintainers to be repository maintainers?  Currently you don't.

> Yes, Flux also uses git repos for non-code contributions (fluxcd/community, fluxcd/website which also includes docs). The community role of [Project Member](https://github.com/fluxcd/community/blob/main/community-roles.md#project-member) currently says:
> 
> > "Project Members are [Community Members](https://github.com/fluxcd/community/blob/main/community-roles.md#community-member) who contribute directly to the project and add value to it. This can be through code, documentation, taking part in bug scrubs, etc."

OK, that works then. 

> > * [x]  3. There's no clearly defined process for removing a Maintainer.  Consider that you may someday need to do so becuase of a CoC violation.; there needs to be a process that will work in that case, even if the errant maintainer is hostile to being removed.
> >   See our template for an example of language you can use
> 
> Adopted CoC rules as per suggestion from tag-contributor-strategy: https://github.com/cncf/project-template/blob/main/GOVERNANCE-maintainer.md in [590ff08](https://github.com/fluxcd/community/commit/590ff08ebcd553f9e52577dc3e91f180679d10b2).

Except that CoC violations aren't the only reason to remove a flux2 maintainer.  Inactivity would be another, or simply a failure to contribute.  What I'm referring to is this:

> Involuntary removal is handled by the [flux2 maintainers](https://github.com/fluxcd/community/blob/5c37114ded0ad53edce529eb2fa412955dbe76fa/community-roles.md#flux2-maintainers) team. Removing a Repository Maintainer for any reason other than inactivity is a [Supermajority decision](https://github.com/fluxcd/community/blob/5c37114ded0ad53edce529eb2fa412955dbe76fa/PROCESS.md#supermajority-decisions).

The wording there suggests that there is no way to involuntarily remove a flux2 maintainer.  You can fix that with:

`Removing a Repository Maintainer or flux2 Maintainer for any reason ...`

> 
> > * [x]  4. Unanimous votes: You may not have considered the difficulty of holding unanimous votes with a 7-person committee.  Consider the difficultly of holding a vote if someone is out-of-office.  I think this made more sense with a 3-person group.  I suggest a 2/3 majority instead.
> 
> Removed the unanimity option to prevent a potential future impasse. Decisions that previously required unanimity now only require supermajority. In [86b50cb](https://github.com/fluxcd/community/commit/86b50cb6f52592cb3faad06a8a4108e6d3738ffb).
> 
> > * [ ]  5. Your other vote mechanisms are defined by a link to wikipedia.  The wikipedia links do not actually define how these votes would be counted in practice.  For example, does one need a simple majority of all maintainers in existence, or just of the maintainers voting?   If the latter, what's the notification process and the time limit?
> 
> Replaced those with a definition of Simple Majority and Supermajority in [5c37114](https://github.com/fluxcd/community/commit/5c37114ded0ad53edce529eb2fa412955dbe76fa).
> 
> question @jberkus does this read more clearly?

Let me suggest:

`... more than half of all Deciders`

and 

`... two-thirds or more of all Deciders`

This removes any ambiguity as to who you need to count.  It says that decisions need to be counted on the basis of all (relevant) Deciders in existence, not just all Deciders voting, which then removes the necessity to document notification and vote timing intervals, which lessens overhead and allows you to use flexible decision-making mechanisms.

> 
> If consensus can not be reached, and a vote is called, the intention is to require either a simple or supermajority of the deciders (depending on the decision, as noted in the "Deciders" section of the doc), not only of those who cast their vote. We did not yet specify the notification process and time limit for decisions generally, because it's not the latter.
> 
> question @jberkus In practice, Flux maintainers use the public community meetings to inform about decisions in progress and decisions already made. Do you think we should explicitly note this in the governance doc, and/or adopt a different notification process?

If that process already exists, you should mention it.  It reassures new contributors that there is a transparent process.

Overall looks good.

dholbach commented:
@jberkus Thanks again for all the feedback. @scottrigby and I went over this one more time, collected feedback from the maintainers and feel it's ready for review now.

We thought that the next steps will be:

1. Discuss in next [Flux Dev meeting](https://fluxcd.io/community/#meetings) (tomorrow)
2. Start discussion (and vote) on the PR
3. Once concluded and accepted, merge
4. Then file follow-up issues to implement relevant in GitHub and elsewhere

Thanks again for everyone's feedback so far - this whole effort will bring more clarity and ease of interaction to our community! <3

hiddeco commented:
> The idea was brought forward that "flux2 maintainers" might be clear to us as the Flux community, but as with this governance change, there are also admin tasks involved like creating repos, etc the question was brought up if we shouldn't use a new term for this team, e.g. "flux admins" or "core maintainer".

I think this is a good suggestion, and would opt for the latter if no other alternatives are suggested.

stefanprodan commented on GOVERNANCE.md:
```diff
@@ -28,76 +27,72 @@ Flux strives to operate independently of specific partisan interests, and for de
 
 - Repository [Maintainers][Maintainer]: Decisions that affect only one git repository.
-- [Oversight Committee]: Decisions that are outside the scope of a single git repository.
+- [flux2 maintainers]: Decisions that are outside the scope of a single git repository.
```
I would change the name to **core maintainers** since the members of this group will have admin privileges of the whole organization and all Flux related projects.

dholbach replied:
Are we going to create a team called https://github.com/orgs/fluxcd/teams/flux2-maintainers ... or rename it?

stefanprodan replied:
I would rename the current one, after this PR is merged.

hiddeco commented on community-roles.md:
```diff
@@ -96,7 +96,7 @@ This team drives the direction, values and governance of the overall project.
 Ensuring that oversight of the project is not controlled by one company or organization.
 
-**Defined by:** entry in [flux2 MAINTAINERS file](https://github.com/fluxcd/flux2/blob/main/MAINTAINERS), and in the `@fluxcd/flux2-maintainers` GitHub team.
+**Defined by:** entry in [flux2 MAINTAINERS file](https://github.com/fluxcd/flux2/blob/main/MAINTAINERS), and in the `@fluxcd/core-maintainers` GitHub team.
```
I would like to see this file moved to a neutral place.

dholbach replied:
`https://github.com/fluxcd/community/blob/main/CORE-MAINTAINERS` maybe?

dholbach replied:
And remove `https://github.com/fluxcd/flux2/blob/main/MAINTAINERS)` then?

stefanprodan replied:
We can't move it, each repo must contain a MAINTAINERS file, but we can link from flux2 MAINTAINERS file to community CORE-MAINTAINERS one.

dholbach replied:
Ok.

dholbach replied:
:+1:

dholbach commented on PROCESS.md:
This was just copied/pasted from the previous incarnation of the doc, but you are right. Let me improve this somewhat.

makkes commented on PROCESS.md:
```suggestion
Org Admins are defined in the [ORG-ADMINS file](https://github.com/fluxcd/community/blob/main/ORG-ADMINS). An election can be done through a pull request against this file to be approved by the [core maintainers].
```

makkes commented on PROCESS.md:
How would one apply for core maintainership?

stefanprodan commented on ORG-ADMINS:
```diff
@@ -1,7 +1,10 @@
 | -- | -- | -- | -- |
+| Chris Aniszczyk | `caniszczyk` | `@caniszczyk` | CNCF |
 | Hidde Beydals | `hidde` | `@hiddeco` | Weaveworks  |
+| Ihor Dvoretskyi | `Ihor Dvoretskyi` | `idvoretskyi` | CNCF |
```
Ihor has no admin access to the org, it's just Chris and LF.

dholbach commented:
Can fluxcd/maintainers please take a look and review?

dholbach commented:
Our current Governance says:

> ## Supermajority Decisions
> 
> [Oversight Committee](https://github.com/fluxcd/community/blob/main/community-roles.md#oversight-committee): Material changes to the Governance document.

Still I'd like to get input from as many maintainers as possible on this, hence pinging everyone: @alisondy @bia @darkowlzz @dinosk @foot @hiddeco @jpellizzari @kingdonb @makkes @mathetake @phillebaba @pjbgf @relu @scottrigby @seaneagan @somtochiama @souleb @squaremo @stefanprodan @yiannistri

(Other community members are obviously also very welcome giving feedback here. 💖)

stefanprodan reviewed (approved):
LGTM 💯 

Thanks to everyone that helped shaping the Flux governance! As a member of the former @fluxcd/oversight-committee I'm very happy to see it replaced will a broader group of maintainers.

hiddeco commented on GOVERNANCE.md:
```diff
@@ -28,75 +27,72 @@ Flux strives to operate independently of specific partisan interests, and for de
-## Roles in the Flux Community
+Maintainers will also have closed meetings in order to discuss security reports
+or Code of Conduct violations.  Such meetings should be scheduled by any
+Maintainer on receipt of a security issue or CoC report.  All current Maintainers
+must be invited to such closed meetings, except for any Maintainer who is
+accused of a CoC violation.
```
Security related issues are often initially only shared on a need-to-know basis, and this process will become even more complicated when we finalize the plans we are currently working on to formalize the Coordinated Vulnerability Disclosure process (likely based on https://github.com/kubernetes/committee-security-response/blob/main/private-distributors-list.md).

dholbach replied:
The above paragraph was copied https://github.com/cncf/project-template/blob/main/GOVERNANCE-maintainer.md#meetings from and was meant to illustrate as to why internal/closed meetings might happen.

Is it OK, if I refer to https://fluxcd.io/security/ for more information on the specifics of the security process to resolve any ambiguity in that paragraph?

hiddeco replied:
Yes, this sounds good to me.

dholbach replied:
:+1:

hiddeco commented on GOVERNANCE.md:
```diff
@@ -28,75 +27,72 @@ Flux strives to operate independently of specific partisan interests, and for de
+
 ### Deciders
 
 - Repository [Maintainers][Maintainer]: Decisions that affect only one git repository.
```
```suggestion
- Repository [Maintainers][Maintainer]: Decisions that affect only one Git repository.
```

(This has been bugging me for some time now. There might be more outside of diff).

hiddeco commented on GOVERNANCE.md:
```diff
@@ -28,75 +27,72 @@ Flux strives to operate independently of specific partisan interests, and for de
 
 - Repository [Maintainers][Maintainer]: Decisions that affect only one git repository.
-- [Oversight Committee]: Decisions that are outside the scope of a single git repository.
+- [core maintainers]: Decisions that are outside the scope of a single git repository.
```
```suggestion
- [core maintainers]: Decisions that are outside the scope of a single Git repository.
```

hiddeco commented on community-roles.md:
```suggestion
Maintainers are elected [Project Members][Project Member] who have shown significant and sustained contributions in a Git repository.
```

hiddeco commented on PROCESS.md:
```suggestion
- If a Maintainer of an affected Git repository feels feedback from specific people is warranted they will @mention those users or teams to request feedback.
```

hiddeco commented on CORE-MAINTAINERS:
```diff
@@ -0,0 +1,21 @@
+https://cloud-native.slack.com in #flux (https://cloud-native.slack.com/messages/CLAJ40HV3)
+(obtain an invitation at https://slack.cncf.io/).
+
+These maintainers are shared with other Flux v2-related git
```
```suggestion
These maintainers are shared with other Flux v2-related Git
```

hiddeco commented on community-roles.md:
```suggestion
- Serve as a point of conflict resolution between Contributors to their Git repository
```

hiddeco commented on community-roles.md:
```suggestion
In addition to maintaining `flux2` and Flux 2-related repositories, this team serves as escalation point for the overall project, and anything not easily managed by the Maintainers of each Git repository.
```

hiddeco commented on community-roles.md:
```diff
@@ -162,53 +112,26 @@ The following apply to all assets across the Flux org:
 - Defining the scope of each git repository
```
```suggestion
- Defining the scope of each Git repository
```

hiddeco reviewed (commented):
This looks good to me, only have minor nitpicks left. :sun_with_face:

hiddeco reviewed (approved):
Happy with the outcome of this, thanks for putting the icing on the cake @dholbach :cake:

darkowlzz commented on GOVERNANCE.md:
```diff
@@ -28,75 +27,74 @@ Flux strives to operate independently of specific partisan interests, and for de
+See
+
+- [community-roles.md] for the definition of roles in our community and
+- [PROCESS.md] to understand how to interact with Flux Community processes
```
Looks like this is missing a link at the bottom of the page, due to which, in the markdown preview PROCESS.md is not a hyperlink.

dholbach replied:
:+1:

darkowlzz commented on community-roles.md:
```suggestion
  This can be evidenced differently and we want to maintain some general flexibility assessing this. Significant and sustained contributions can be both by showing a long-term level of care with a bigger number of smaller contributions or by a smaller set of sizable contributions. To make it somewhat more comparable, here is an example of commitment we would be happy to accept for a maintainer:
```

pjbgf reviewed (approved):
Phenomenal work @dholbach and @scottrigby ! :clap: :clap:

darkowlzz reviewed (approved):
LGTM!
Thanks.

kingdonb commented on PROCESS.md:
```suggestion
File an issue in the `fluxcd/community` repository and contact the `@fluxcd/org-admins` team to fulfill your request.
```

scottrigby replied:
I believe this is due to British English and US English spelling differences 🇬🇧 🙃 🇺🇸

scottrigby replied:
But still, no harm, so merging. Thanks @kingdonb 😊

kingdonb reviewed (approved):
👍 with one spelling nit

scottrigby commented on ORG-ADMINS:
```diff
@@ -1,7 +1,9 @@
+| Chris Aniszczyk | `caniszczyk` | `@caniszczyk` | CNCF |
 | Hidde Beydals | `hidde` | `@hiddeco` | Weaveworks  |
 | Stefan Prodan | `stefanprodan` | `@stefanprodan` | Weaveworks  |
 | Michael Bridgen | `Michael Bridgen` | `@squaremo` | Weaveworks  |
+| The Linux Foundation | - | `thelinuxfoundation` | The Linux Foundation |
```
I remember @dholbach said @squaremo was OK with being removed in the earlier draft of this PR (when `fluxcd/flux2` repo maintainers were proposed as the project escalation point). 

However, now this PR now has separated Core Maintainers from `fluxcd/flux2` repo maintainers, and there is no longer a requirement that Core maintainers be listed in other `fluxcd` GitHub org repo MAINTAINERS files. Instead it's the other way around, granting implicit maintainership in other Flux repos where listed:

> Maintainership in the [CORE-MAINTAINERS file](https://github.com/fluxcd/community/blob/main/CORE-MAINTAINERS) trickles down to all other Flux 2-related repositories which means that maintainers mentioned there are also maintainers in all other repositories.
>
> In addition to maintaining `flux2` and Flux 2-related repositories, this team serves as escalation point for the overall project, and anything not easily managed by the Maintainers of each Git repository.

Therefore @squaremo could remain as one of the Flux Core Maintainers, which I would love to see.

dholbach replied:
No, this is not quite right.

We have

1. The Core Maintainers, which are maintainers of Flux2. They are escalation point for a lot of questions - as layed out in this PR.
2. The Org Admins. These are folks that have elevated access to the Flux org in GH. Not everyone in the Flux2 maintainers group is comfortable with having them all. The `ORG-ADMINS` team you commented on just documents the status quo (adding CNCF and LF, so everyone who has access - this is fully transparent now).

See the list of follow-up items on the PR at the top: we will a) review the list after this PR is merged and see if we have new applicants and b) revisit the team annually.

makkes replied:
I wonder whether we should add some kind of a summary to the README or the PROCESS document that just lays out the different groups we have defined: org admins, core maintainers, individual repo maintainers, ..., each accompanied by a one-sentence description and link to the detailed description of the particular group's responsibilities etc.

dholbach replied:
https://github.com/dholbach/community/blob/disolve-oversight-committee/community-roles.md#roles would be a great place.

Let's create the Org Admin team in GH once this is approved and add it to the list. It's right at the top so should be self explanatory - wdyt?

scottrigby reviewed (approved):
I'm really happy with @jberkus' involvement in feedback, and how this governance change proposal turned out. Kudos to everyone involved 👏 +1 from me. 

@squaremo when you review this, just want to be sure that you're signing on for the Core Maintainer role, unlike the earlier draft – see inline comment below. Tl;dr, totally legit if you want to and I hope you do. If so, this all looks perfect to me as-is 💖

squaremo commented:
> Removed the unanimity option to prevent a potential future impasse. Decisions that previously required unanimity now only require supermajority. In [86b50cb](https://github.com/fluxcd/community/commit/86b50cb6f52592cb3faad06a8a4108e6d3738ffb).

I don't like this change. It doesn't fit with the spirit of consensus-seeking, and I don't think it is effective at preventing impasses. The difference between requiring a unanimous vote and requiring a majority vote is that someone abstaining can prevent a unanimous vote from being counted. But in the original formulation, voting is a last resort that everyone needs to agree to -- so if you were seeking to block a decision, you would simply refuse to agree to a vote.

squaremo commented:
I'm happy with the more precise language, and bringing things in line with CNCF advice (thanks @jberkus!). I'm also happy to 1. be removed from the decision-making body (core maintainers) and 2. remain in the executive body (org admins) for now.

Regarding unanimity v super-majorities, which I commented on above: that isn't a blocker for me, I just don't think it has the effect claimed, and deserves more consideration.

dholbach commented:
> I don't like this change. It doesn't fit with the spirit of consensus-seeking, and I don't think it is effective at preventing impasses. The difference between requiring a unanimous vote and requiring a majority vote is that someone abstaining can prevent a unanimous vote from being counted. But in the original formulation, voting is a last resort that everyone needs to agree to -- so if you were seeking to block a decision, you would simply refuse to agree to a vote.

The idea behind this is primarily to avoid situations where a decisions can't be made because e.g. one core maintainer is on extended holidays or finds it hard to take the time to respond.

I agree that consensus seeking is very much the spirit of what we do and have done all the time and while it's already documented in this PR, maybe an extra clause that explains that a veto votes given in a specific time frame would essentially mean mean "more discussion is needed". Would that help?

dholbach commented:
> I'm happy with the more precise language, and bringing things in line with CNCF advice (thanks @jberkus!). I'm also happy to 1. be removed from the decision-making body (core maintainers) and 2. remain in the executive body (org admins) for now.
> 
> Regarding unanimity v super-majorities, which I commented on above: that isn't a blocker for me, I just don't think it has the effect claimed, and deserves more consideration.

Thanks a lot for the feedback and all your help so far!

squaremo commented:
> a veto votes given in a specific time frame would essentially mean mean "more discussion is needed". Would that help?

People can already keep discussion running just by saying that's what they want. The exception given is if there's a deadline and agreement hasn't been achieved; the situation you're trying to avoid is this:

 - there's a deadline for some crucial decision that needs everyone to agree;
 - but someone can't be reached, so you can't claim everyone has agreed, and the deadline passes without action.

Requiring only two-thirds majority means you only need two-thirds of people to cast a vote, assuming they all agree, so as long as you have three or more people involved, you can get the decision made if an individual is unreachable. (You could still have the same problem, if more than a third of the people can't be reached -- it's just less likely.)

There's no impasse here, just a decision that isn't made on time. Things like electing a new maintainer are not that likely to have a deadline necessitating a vote, so there isn't a problem to be solved for those -- on the other hand, how would you feel if you became a maintainer despite some of the people you'll be working with voting against it.

squaremo reviewed (approved):
Approved on the basis that I am happy to retire from my oversight role, and after that you can change what you like anyway :-D

phillebaba reviewed (approved):
LGTM! Seems like a step in the right direction.

dholbach commented:
I addressed the comments by @squaremo by a suggestion of his, which I found sensible: instead of removing unanimity votes for a subset of decisions altogether, instead allow us to require specifying a deadline, quorum of 2/3 and no veto votes instead.

This is very closely modeled on what we have been practicing so far, avoids impasse and encourages building consensus - just how we like it.

@makkes and @hiddeco were supportive of this new clause as well.

As we have all the votes of the former Flux Oversight Committee and more than 2/3 of all sub-project maintainers already, I'll give this a couple more days for folks to have their say. Then let's merge this and announce the governance change. How about mid next-week?

dholbach commented:
+1 on all of this from my end as well. (Signalling this as I can't :heavy_check_mark: my own PR. :-))

scottrigby reviewed (approved):
I caught up on the slack thread about changes since last review. Looks good!

dholbach commented:
Thank you very much everyone for this great team-work. Defining governance and process is not easy and I feel together with the help of the CNCF community we struck a really fine balance between clarity, consensus building, documenting the status quo and making sure our community can continue to grow.

I'm happy and proud to be part of this great project and look forward to where we are taking it further together. :heart: 

I will file issues to resolve the follow-up items as a next step.
